src/shapes/rect.class.ts

```diff
@@ -1,179 +1,180 @@
+    // with caching and higher zoom level this makes more damage than help
+    let { rx, ry } = this;
+    const { width: w, height: h } = this;
+    const x = -w / 2;
+    const y = -h / 2;
+    rx = rx ? Math.min(rx, w / 2) : 0;
+    ry = this.ry ? Math.min(ry, h / 2) : 0;
+    const isRounded = rx !== 0 || ry !== 0;
+
+    ctx.beginPath();
+
+    ctx.moveTo(x + rx, y);
+
+    ctx.lineTo(x + w - rx, y);
+    isRounded && ctx.bezierCurveTo(x + w - kRect * rx, y, x + w, y + kRect * ry, x + w, y + ry);
+
+    ctx.lineTo(x + w, y + h - ry);
+    isRounded && ctx.bezierCurveTo(x + w, y + h - kRect * ry, x + w - kRect * rx, y + h, x + w - rx, y + h);
+
+    ctx.lineTo(x + rx, y + h);
+    isRounded && ctx.bezierCurveTo(x + kRect * rx, y + h, x, y + h - kRect * ry, x, y + h - ry);
+
+    ctx.lineTo(x, y + ry);
+    isRounded && ctx.bezierCurveTo(x, y + kRect * ry, x + kRect * rx, y, x + rx, y);
+
+    ctx.closePath();
+
+    this._renderPaintInOrder(ctx);
+  },
 
   /**
-   * Returns {@link fabric.Rect} instance from an object representation
-   * @static
-   * @memberOf fabric.Rect
-   * @param {Object} object Object to create an instance from
-   * @returns {Promise<fabric.Rect>}
+   * Returns object representation of an instance
+   * @param {Array} [propertiesToInclude] Any properties that you might want to additionally include in the output
+   * @return {Object} object representation of an instance
    */
-  fabric.Rect.fromObject = function(object) {
-    return fabric.Object._fromObject(fabric.Rect, object);
-  };
+  toObject: function(propertiesToInclude) {
+    return this.callSuper('toObject', ['rx', 'ry'].concat(propertiesToInclude));
+  },
 
-})(typeof exports !== 'undefined' ? exports : window);
+  /* _TO_SVG_START_ */
+  /**
+   * Returns svg representation of an instance
+   * @return {Array} an array of strings with the specific svg representation
+   * of the instance
+   */
+  _toSVG: function() {
+    const { width, height, rx, ry } = this;
+    return [
+      '<rect ', 'COMMON_PARTS',
+      'x="', -width / 2, '" y="', -height / 2,
+      '" rx="', rx, '" ry="', ry,
+      '" width="', width, '" height="', height,
+      '" />\n'
+    ];
+  },
+  /* _TO_SVG_END_ */
+});
+
+/* _FROM_SVG_START_ */
+/**
+ * List of attribute names to account for when parsing SVG element (used by `Rect.fromElement`)
+ * @static
+ * @memberOf Rect
+ * @see: http://www.w3.org/TR/SVG/shapes.html#RectElement
+ */
+Rect.ATTRIBUTE_NAMES = fabric.SHARED_ATTRIBUTES.concat('x y rx ry width height'.split(' '));
+
+/**
+ * Returns {@link Rect} instance from an SVG element
+ * @static
+ * @memberOf Rect
+ * @param {SVGElement} element Element to parse
+ * @param {Function} callback callback function invoked after parsing
+ * @param {Object} [options] Options object
+ */
+Rect.fromElement = function(element, callback, options = {}) {
+  if (!element) {
+    return callback(null);
+  }
+  const {
+    left = 0,
```

Comment:
this highligtherd an issue of this code.
visible needs to be default true. Because ti requires the knowledge of the fact that the attribute default value on SVG is true. It broke my tests for long, i tought it was an es6 issue, while it was just a real code change i introduced without knowing.

Reply:
visible false is hidden, visible undefined means visible.